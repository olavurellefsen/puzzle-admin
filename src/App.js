import React from 'react'
import Menu from './Menu/Menu'
import Home from './Home/Home'
import { ApolloProvider } from 'react-apollo'
import makeApolloClient from './Hasura/apollo'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import PrivateRoute from './Routing/PrivateRoute'
import { useAuth0 } from './Auth/react-auth0-wrapper'
import styled from 'styled-components'

export const AppContainer = styled.div`
  display: flex;
  flex-direction: row;
  overflow: hidden;
  height: 100%;
  background-color: #eeeeee;
`
const client = makeApolloClient();

function App() {
  const { loading } = useAuth0()

  return (
    <div className="App">
      {!loading && (
        <BrowserRouter>
          <ApolloProvider client={client}>
            <AppContainer>
              <Menu />
              <Switch>
                <Route path="/" exact component={Home} />
                <PrivateRoute path="/home" component={Home} />
              </Switch>
            </AppContainer>
          </ApolloProvider>
        </BrowserRouter>
      )}
    </div>
  )
}

export default App
