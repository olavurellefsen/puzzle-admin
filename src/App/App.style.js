import styled from 'styled-components';

export const AppContainer = styled.div`
  color: red;
`;
AppContainer.displayName = 'AppContainer';
