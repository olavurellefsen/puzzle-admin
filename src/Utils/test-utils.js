import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
