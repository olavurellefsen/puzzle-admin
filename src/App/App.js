import React from "react";
import { AppContainer } from "./App.style";
import Scenes from "../Scenes/Scenes";

const App = () => (
  <AppContainer>
    <Scenes />
  </AppContainer>
);

export default App;
