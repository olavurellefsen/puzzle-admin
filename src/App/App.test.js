import React from "react";
import { render, cleanup } from 'test-utils';
import App from "./App";
afterEach(cleanup)

const auth = { isAuthenticated: () => false}
it("should render correctly", () => {
  const { container } = render(<App auth={auth} />);
  expect(container).toMatchSnapshot();
});

it("should show the words Puzzle Admin System", () => {
  const { getByText } = render(<App auth={auth} />);
  expect(getByText("Puzzle Admin System")).toBeTruthy();
});