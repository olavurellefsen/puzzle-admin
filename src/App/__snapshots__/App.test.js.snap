// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa crgcgU"
  >
    <div
      class="sc-bwzfXH brfdCc"
    >
      <img
        class="sc-htpNat hFgGyf"
        src="images/logo.png"
      />
      <div
        class="sc-bxivhb DwAvV"
      >
        Puzzle Admin System
      </div>
      <div
        class="sc-ifAKCX dEMvjk"
      >
        <div
          class="sc-EHOje kIVrKf"
        >
          Login
        </div>
      </div>
    </div>
  </div>
</div>
`;
